pkg/syncer/util.go

```diff
@@ -623,3 +625,89 @@ func getPVsInBoundAvailableOrReleasedForVc(ctx context.Context, metadataSyncer *
 
 	return k8svolumes, nil
 }
+
+// createVolumeOnMultiVc attemts to create a static volume on each VC until it gets SUCCESS.
+// If while creating the volume, CNS returns CnsAlreadyRegisteredFault, it means that the volume does not need to be re-created.
```

Comment:
where are you handling `CnsAlreadyRegisteredFault`?

Reply:
It is being taken of by CNS Create Volume. Check here: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/common/cns-lib/volume/util.go#L287

However, there is a bug on CNS side where they are not returning `CnsAlreadyRegisteredFault ` for file volumes. I have raised a bug against them. Bug #3297537